src/app/routes/cpsAsset/getInitialData/addMpuBlock/index.js

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable prefer-destructuring */
+import path from 'ramda/src/path';
+import pathOr from 'ramda/src/pathOr';
+import deepClone from 'ramda/src/clone';
+import splitAt from 'ramda/src/splitAt';
+import { getNthCpsParagraphIndex } from '../helpers';
+
+const mpuBlock = {
+  type: 'mpu',
+  model: {},
+};
+
+const addMpuBlock = json => {
+  const pageData = deepClone(json);
+  const pageType = path(['metadata', 'type'], pageData);
+  const { allowAdvertising } = path(['metadata', 'options'], pageData);
+  const blocks = pathOr([], ['content', 'model', 'blocks'], pageData);
+
+  if (pageType !== 'STY' || !blocks || !allowAdvertising) {
```

Comment:
Please can we:
- Do the `allowAdvertising` first, so we can short-circuit ahead of time (before using `deepClone` etc).
- Fix (remove?) the `!blocks` condition. We're using pathOr to create an array, so blocks will never be falsy.

Reply:
Thanks for making these changes, @rhenshaw56.

Reply:
done.